packages/react-components/react-combobox/src/components/Combobox/useCombobox.tsx

```diff
@@ -156,31 +80,11 @@ export const useCombobox_unstable = (props: ComboboxProps, ref: React.Ref<HTMLIn
     },
     elementType: 'input',
   });
-  const resolvedPropsOnKeyDown = triggerSlot.onKeyDown;
-  triggerSlot.onChange = mergeCallbacks(triggerSlot.onChange, onTriggerChange);
-  triggerSlot.onBlur = mergeCallbacks(triggerSlot.onBlur, onTriggerBlur); // only resolve listbox slot if needed
-  listboxSlot =
-    open || hasFocus
-      ? slot.optional(props.listbox, {
-          renderByDefault: true,
-          defaultProps: { children: props.children },
-          elementType: Listbox,
-        })
-      : undefined;
-  [triggerSlot, listboxSlot] = useTriggerListboxSlots(props, baseState, ref, triggerSlot, listboxSlot);
-  const listboxRef = useMergedRefs(listboxSlot?.ref, comboboxPopupRef);
-
-  if (hideActiveDescendant) {
-    triggerSlot['aria-activedescendant'] = undefined;
-  }
-
-  if (listboxSlot) {
-    listboxSlot.ref = listboxRef;
-  }
+  triggerSlot = useInputTriggerSlot(props, baseState, ref, triggerSlot);
```

Comment:
Why do we have to pass `ref`, can we do:
on L75: `ref: useMergedRefs(props.input?.ref, triggerRef, ref)`,
and in `useInputTriggerSlot` just use triggerSlot.ref?

Reply:
Yeah also, the `ref` is already in the slot at that point - there's no point even having a ref argument either